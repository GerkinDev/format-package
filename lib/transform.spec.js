const transform = require('./transform');

describe('transform', () => {
  it('alphabetizes the object if no transformation is provided', () => {
    const key = 'foo';
    const value = { qux: 'baz', bar: 'foo' };
    expect(transform(key, value)).toMatchSnapshot();
  });

  it('uses the transformation by key', () => {
    const key = 'foo';
    const value = 'prev';
    const transformations = {
      foo: jest.fn(k => [k, 'transformed']),
    };
    expect(transform(key, value, transformations)).toMatchSnapshot();
    expect(transformations.foo).toHaveBeenCalledWith(key, value);
  });

  it('returns the untransformed values otherwise', () => {
    const key = 'foo';
    const value = 'bar';
    expect(transform(key, value)).toEqual([key, value]);
  });
});
