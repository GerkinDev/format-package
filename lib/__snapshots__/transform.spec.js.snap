// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform alphabetizes the object if no transformation is provided 1`] = `
Array [
  "foo",
  Object {
    "bar": "foo",
    "qux": "baz",
  },
]
`;

exports[`transform uses the transformation by key 1`] = `
Array [
  "foo",
  "transformed",
]
`;
